### PR TITLE
[mds-agency] [mds-cache] [mds-logger] [mds-stream] JSON.stringify logs, add some redundancy to failed cache/stream writes

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
Tweak mds-logger to emit stringified JSON logs so they can be properly parsed by Kibana, add some more redundancy to failed cache/stream calls when they're client disconnected errors, add more logs all over.

## 📦 Impacts:
- mds-agency
- mds-cache
- mds-logger
- mds-stream

### CODE REVIEW

File: packages/mds-stream/kafka/agency-stream-kafka.ts [hunk 1]
```diff
@@ -36,10 +36,46 @@ export const AgencyStreamKafka: AgencyStreamInterface = {
       tripMetadataProducer.initialize()
     ])
   },
-  writeEvent: eventProducer.write,
-  writeTelemetry: telemetryProducer.write,
-  writeDevice: deviceProducer.write,
-  writeTripMetadata: tripMetadataProducer.write,
+  writeEvent: async msg => {
+    try {
```
Comment (drtyh2o) [inline packages/mds-stream/kafka/agency-stream-kafka.ts:40]: I am wondering if there’s a way to handle this pattern generically within the `write` method.

**Review by drtyh2o — APPROVED** [review]

File: packages/mds-stream/kafka/agency-stream-kafka.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-stream/kafka/agency-stream-kafka.ts:40]: Yeah, most likely, I was a little spooked about doing some weird self mutation within the write method itself though

File: packages/mds-agency-cache/index.ts [hunk 2]
```diff
@@ -424,7 +434,12 @@ async function writeOneTelemetry(telemetry: Telemetry) {
 }
 
 async function writeTelemetry(telemetries: Telemetry[]) {
-  await Promise.all(telemetries.map(telemetry => writeOneTelemetry(telemetry)))
+  try {
+    await Promise.all(telemetries.map(telemetry => writeOneTelemetry(telemetry)))
+  } catch (err) {
+    logger.error('Failed to write to cache', err)
```
Comment (BigIdeas) [inline packages/mds-agency-cache/index.ts:440]: nit - no need to identify that it was Telemetry that couldn't be written?

**Review by BigIdeas — APPROVED** [review]

File: packages/mds-agency-cache/index.ts (hunk 2 above)
Comment (avatarneil) [reply packages/mds-agency-cache/index.ts]: Good catch, fixed in f1406ab

**Pull request merged**